src/test/objectstore/store_test.cc

```diff
@@ -903,6 +903,205 @@ TEST_P(StoreTest, CompressionTest) {
   g_ceph_context->_conf->apply_changes(NULL);
 }
 
+TEST_P(StoreTest, garbageCollection) {
```

Comment:
The tricky thing I found was: even reading the data and verifying them also does not say anything about  GC run and its correctness .  
 To do testing:  I used to debug print the "blob_depth" of the lextents before and after doing GC and used to manually verify them.
 
Let me try to use one of your other suggestion regarding adding a new stats counter  ".... space saved by GC procedure" to do the UT.